### PR TITLE
wrap: init at 0.3.1

### PR DESCRIPTION
**Pull request by austinbutler**
###### Motivation for this change

Add package. It's somewhat difficult to get tools like this to use the right font with Nix.

Initializing at unstable because I submitted a PR that helps the program find the font and that was merged, but there hasn't been a release yet. I'll update to a version when there is.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
OfBorg complains

> while evaluating anonymous function at <…>/pkgs/applications/radio/gnuradio/wrapper.nix:1:1
> value is a set while a Boolean was expected

because the new package name `wrap` conflicts with this boolean-only input to gnuradio:

https://github.com/NixOS/nixpkgs/blob/8e4fe32876ca15e3d5eb3ecd3ca0b224417f5f17/pkgs/applications/radio/gnuradio/wrapper.nix#L5-L7

I think it's better to rename that input to e.g. `doWrap` or `withWrapping`. Thoughts @doronbehar?

**Comment by doronbehar** [comment]
> I think it's better to rename that input to e.g. `doWrap` or `withWrapping`. Thoughts @doronbehar?

Agree. But for explicitly, if you could please also do:

```diff
diff --git i/pkgs/applications/radio/gnuradio/wrapper.nix w/pkgs/applications/radio/gnuradio/wrapper.nix
index d255b199bc9..378efb77fb8 100644
--- i/pkgs/applications/radio/gnuradio/wrapper.nix
+++ w/pkgs/applications/radio/gnuradio/wrapper.nix
@@ -4,7 +4,7 @@
 , unwrapped
 # If it's a minimal build, we don't want to wrap it with lndir and
 # wrapProgram..
-, wrap ? true
+, doWrap ? true
 # For the wrapper
 , makeWrapper
 # For lndir
@@ -138,7 +138,7 @@ let
     ;
     pkgs = packages;
   };
-  self = if wrap then
+  self = if doWrap then
     stdenv.mkDerivation {
       inherit name passthru;
       buildInputs = [
diff --git i/pkgs/top-level/all-packages.nix w/pkgs/top-level/all-packages.nix
index 76c68e6b395..3458d02382b 100644
--- i/pkgs/top-level/all-packages.nix
+++ w/pkgs/top-level/all-packages.nix
@@ -22925,6 +22925,7 @@ in
   gnss-sdr = callPackage ../applications/radio/gnss-sdr { };
 
   gnuradio = callPackage ../applications/radio/gnuradio/wrapper.nix {
+    doWrap = true;
     unwrapped = callPackage ../applications/radio/gnuradio {
       inherit (darwin.apple_sdk.frameworks) CoreAudio;
       python = python3;
@@ -22935,7 +22936,7 @@ in
   # A build without gui components and other utilites not needed for end user
   # libraries
   gnuradioMinimal = gnuradio.override {
-    wrap = false;
+    doWrap = false;
     unwrapped = gnuradio.unwrapped.override {
       volk = volk.override {
         # So it will not reference python
@@ -22955,6 +22956,7 @@ in
     };
   };
   gnuradio3_8 = callPackage ../applications/radio/gnuradio/wrapper.nix {
+    doWrap = true;
     unwrapped = callPackage ../applications/radio/gnuradio/3.8.nix {
       inherit (darwin.apple_sdk.frameworks) CoreAudio;
       python = python3;
@@ -22965,7 +22967,7 @@ in
   # A build without gui components and other utilites not needed if gnuradio is
   # used as a c++ library.
   gnuradio3_8Minimal = gnuradio3_8.override {
-    wrap = false;
+    doWrap = false;
     unwrapped = gnuradio3_8.unwrapped.override {
       volk = volk.override {
         enableModTool = false;
@@ -22983,6 +22985,7 @@ in
     };
   };
   gnuradio3_7 = callPackage ../applications/radio/gnuradio/wrapper.nix {
+    doWrap = true;
     unwrapped = callPackage ../applications/radio/gnuradio/3.7.nix {
       inherit (darwin.apple_sdk.frameworks) CoreAudio;
       python = python2;
@@ -22994,7 +22997,7 @@ in
   # A build without gui components and other utilites not needed if gnuradio is
   # used as a c++ library.
   gnuradio3_7Minimal = gnuradio3_7.override {
-    wrap = false;
+    doWrap = false;
     unwrapped = gnuradio3_7.unwrapped.override {
       volk = volk.override {
         enableModTool = false;
```

**Comment by austinbutler** [comment]
Made the changes, got the green check mark :heavy_check_mark:.

**Comment by doronbehar** [comment]
@austinbutler could please put the gnuradio related changes in a separate commit, before the `wrap: init` commit, and explain in the commit message that this was suggested here.

**Comment by austinbutler** [comment]
@doronbehar done.

File: pkgs/tools/text/wrap/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub, makeWrapper, courier-prime }:
+
+buildGoModule rec {
+  pname = "wrap";
+  version = "unstable-2021-04-13";
```
Comment (doronbehar) [inline pkgs/tools/text/wrap/default.nix:5]: I think the version can be 0.3.1, with the PR you submitted upstream used here with `fetchpatch` https://github.com/Wraparound/wrap/compare/v0.3.1...develop .

File: pkgs/tools/text/wrap/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Wraparound";
+    repo = "wrap";
+    rev = "384845d1e77207d32c3b3e74694cf26e46cc363d";
+    sha256 = "15dmrm5zcs7yp9lvq630x22ra33yy3rnrbjrh3rzc9nwsyln1prl";
+  };
+
+  buildInputs = [ makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/wrap/default.nix:14]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/tools/text/wrap/default.nix (hunk 1 above)
Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix:5]: We'd watch the fetchpatch pinned to a commit though, right? Not to `develop`?

Like https://github.com/Wraparound/wrap/compare/v0.3.1...b72c280b6eddba9ec7b3507c1f143eb28a85c9c1.patch

Comment (doronbehar) [reply pkgs/tools/text/wrap/default.nix:5]: Use https://github.com/Wraparound/wrap/commit/b72c280b6eddba9ec7b3507c1f143eb28a85c9c1.patch .

File: pkgs/tools/text/wrap/default.nix (hunk 2 above)
Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix]: Changed.

File: pkgs/tools/text/wrap/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/wrap --prefix XDG_DATA_DIRS : ${courier-prime}/share/
+  '';
+
+  meta = with lib; {
+    description = "A Fountain export tool with some extras";
+    homepage = "https://github.com/Wraparound/wrap";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/tools/text/wrap/default.nix:33]: gpl3Only or gpl3Plus?

File: pkgs/tools/text/wrap/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  nativeBuildInputs = [ makeWrapper ];
+
+  vendorSha256 = "03q5a5lm8zj1523gxkbc0y6a3mjj1z2h7nrr2qcz8nlghvp4cfaz";
+
+  patches = [
+    (fetchpatch {
+      name = "courier-prime-variants.patch";
+      url = "https://github.com/Wraparound/wrap/compare/v0.3.1...b72c280b6eddba9ec7b3507c1f143eb28a85c9c1.patch";
+      sha256 = "0cw6vsc37j748kzsyr2w8cx1pmm19z1pij95pk25204yqljbbz84";
+    })
```
Comment (SuperSandro2000) [inline pkgs/tools/text/wrap/default.nix:23]: Actually I think we only need https://github.com/Wraparound/wrap/commit/b72c280b6eddba9ec7b3507c1f143eb28a85c9c1.

File: pkgs/tools/text/wrap/default.nix (hunk 3 above)
Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix:33]: [Project says GPL 3.0](https://github.com/Wraparound/wrap/blob/develop/LICENSE) so I think it's right, or is there some distinction I am missing?

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 121563` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wrap</li>
  </ul>
</details>

File: pkgs/tools/text/wrap/default.nix (hunk 4 above)
Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix]: Yep fixed that.

File: pkgs/tools/text/wrap/default.nix (hunk 1 above)
Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix]: Done!

File: pkgs/tools/text/wrap/default.nix (hunk 3 above)
Comment (doronbehar) [reply pkgs/tools/text/wrap/default.nix:33]: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md

Comment (austinbutler) [reply pkgs/tools/text/wrap/default.nix]: Thanks, that was helpful! I think I have it right now.

**Review by doronbehar — APPROVED** [review]
Diff LGTM and `wrap` passes the `--help` test.

**Pull request merged**